tutorial/opt/aero/plot_optHist.py

```diff
@@ -3,49 +3,92 @@
 import matplotlib.pyplot as plt
 from pyoptsparse import History
 
-plt.rcParams["text.usetex"] = True  # Comment out if latex installation is not present
-plt.rcParams["font.family"] = "serif"
-plt.rcParams["font.size"] = 20
-plt.rcParams["xtick.labelsize"] = 16
-plt.rcParams["ytick.labelsize"] = 16
-plt.rcParams["lines.linewidth"] = 3.0
 
-parser = argparse.ArgumentParser()
-parser.add_argument("--histFile", type=str, default="opt.hst")
-parser.add_argument("--outputDir", type=str, default="./")
-args = parser.parse_args()
+def main():
 
-optHist = History(args.histFile)
-histValues = optHist.getValues()
+    plt.rcParams["text.usetex"] = True  # Comment out if latex installation is not present
+    plt.rcParams["font.family"] = "serif"
+    plt.rcParams["font.size"] = 20
+    plt.rcParams["xtick.labelsize"] = 16
+    plt.rcParams["ytick.labelsize"] = 16
+    plt.rcParams["lines.linewidth"] = 3.0
 
-fig, axes = plt.subplots(nrows=5, sharex=True, constrained_layout=True, figsize=(14, 10))
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--histFile", type=str, default="opt.hst")
+    parser.add_argument("--opt", type=str, default="SLSQP", choices=["IPOPT", "SLSQP", "SNOPT"])
+    parser.add_argument("--outputDir", type=str, default="./")
+    args = parser.parse_args()
 
-axes[0].plot("nMajor", "optimality", data=histValues, label="Optimality")
-axes[0].plot("nMajor", "feasibility", data=histValues, label="Feasibility")
-axes[0].set_yscale("log")
-axes[0].axhline(1e-4, linestyle="--", color="gray")
-axes[0].annotate("Convergence Criteria", xy=(3, 9e-5), ha="left", va="top", fontsize=24, color="gray")
-axes[0].legend(fontsize=20, labelcolor="linecolor", loc="upper right", frameon=False)
-axes[0].autoscale(enable=True, tight=True)
+    try:
```

Comment:
Why do you need this try block? Why can't you just check with `os.path.isfile`?

Reply:
See my comment above, we could check and break out of main() or just return an error, but I honestly don't have strong opinions.
I am more concerned with the failing tests now but this needs a quick cleanup before approval

Reply:
Try/excepts are pretty non-pythonic and we should avoid them if possible. It should just be a simple check of if the file exists, and error out if not.

Reply:
I have removed the try/excepts block for a simple os.path.isfile check